akka-cluster-typed/src/test/scala/docs/akka/cluster/typed/DistributedPubSubExample.scala

```diff
@@ -0,0 +1,359 @@
+          service ! cmd
+          Behaviors.same
+
+      }
+    }
+  }
+}
+
+object DistributedPubSubExample {
+  import akka.actor.testkit.typed.scaladsl.TestProbe
+  import Ontology._
+
+  val config: Config = ConfigFactory.parseString(s"""
+        akka.actor.provider = "cluster"
+        akka.cluster.pub-sub.max-delta-elements = 500
+        akka.cluster.jmx.enabled = off
+        akka.remote.artery.canonical.hostname = 127.0.0.1
+        akka.remote.classic.netty.tcp.port = 0
+        akka.remote.artery.canonical.port = 0
+        akka.loglevel = INFO
+        akka.loggers = ["akka.testkit.TestEventListener"]
+    """)
+
+  def createCluster(nodes: List[ActorSystem[_]]): Unit = {
+    val clusterUp = (nodes: List[ActorSystem[_]], expected: Int) =>
+      nodes.forall { s =>
+        Cluster(s).state.members.count(_.status == MemberStatus.up) == expected
+      }
+
+    val awaitClusterUp = (nodes: List[ActorSystem[_]], expected: Int) =>
+      while (!clusterUp(nodes, expected)) {
+        Thread.sleep(1000)
+      }
+
+    val seed = nodes.head
+    val joinTo = Cluster(seed).selfMember.address
+
+    Cluster(seed).manager ! Join(joinTo)
+    awaitClusterUp(List(seed), 1)
+    nodes.tail.foreach(Cluster(_).manager ! Join(joinTo))
+    awaitClusterUp(nodes, nodes.size)
+  }
+
+  def main(args: Array[String]): Unit = {
+
+    val system = ActorSystem[ProvisionCommand](DataPlatform(), "DataPlatform", config)
+    val system2 = ActorSystem[ProvisionCommand](DataPlatform(), system.name, config)
+    val system3 = ActorSystem[ProvisionCommand](DataPlatform(), system.name, config)
+    val nodes = List(system, system2, system3)
+
+    import akka.actor.typed.scaladsl.adapter._
+    val log = Logging(system.toClassic.eventStream, system.name)
+
+    createCluster(nodes)
+    log.info(s"Cluster up with {} nodes. Starting platform.", nodes.size)
+
+    // provision subscribers
+    val key = "DataTypeA"
+    nodes.foreach(_ ! CreateSubscriber(key))
+
+    // provision new data type
+    val platformProbe = TestProbe[DataApi]()(system)
+    val mediator = DistributedPubSub(system.toClassic).mediator
+    mediator ! DistributedPubSubMediator.Subscribe(IngestionTopic, platformProbe.ref.toClassic)
+
+    system ! ProvisionDataType(key, "dummy-schema", platformProbe.ref)
+    val provisioned = platformProbe.expectMessageType[ProvisionSuccess](5.seconds)
+    log.info(s"Platform provisioned for new data type {}.", provisioned)
+
+    // send and receive in cluster, then for example shutdown
+    val started = platformProbe.expectMessageType[IngestionStarted](5.seconds)
+    log.info(s"Platform ingestion started for {}.", started.key)
+
+    val stopped = platformProbe.expectMessageType[IngestionStopped](5.seconds)
+    log.info(s"Platform ingestion completed for {}.", stopped.key)
+    require(Set(started, stopped).forall(_.key == key))
+
+    nodes.foreach(_.terminate())
+    log.info("Systems terminating.")
+  }
+
+}
```

Comment:
I think this sample is quite a bit too complicated to be a good sample in the docs, it is more like a good sample project to me (or a an example in a book).

Since it is temporary until we have API I guess that is ok though? I think we should seriously simplify and or potentially break this out into a sample project later.